app/src/main/java/com/example/pygmyhippo/user/EventFragment.java

```diff
@@ -90,57 +191,56 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
-                            registerButton.setText("✔");
-                        });
-                        builder.setNegativeButton("No", (DialogInterface.OnClickListener) (dialog, which) -> {
-                            dialog.cancel();
-                        });
-                        AlertDialog geolocationWarning = builder.create();
-                        geolocationWarning.show();
-
-                    } else {
-                        registerButton.setBackgroundColor(0xFF808080);
-                        event.addEntrant(entrant);
-                        registerButton.setText("✔");
-                    }
-                }
+    /**
+     * Register user as an entrant for the current event.
+     *
+     * Button is only visible when the signedInAccount is a user.
+     */
+    private void registerUser() {
+        // if the even already has the user, remove the user upon clicking
+        if (event.hasEntrant(entrant)) {
+            registerButton.setBackgroundColor(0xFF35B35D);
+            event.removeEntrant(entrant);
+            registerButton.setText("Register");
+        } else {
+            // otherwise, check for enabled geolocation and add entrant accordingly
+            if (event.getGeolocation()) {
+
+                AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+                builder.setTitle("WARNING!");
+                builder.setMessage("This event requires geolocation. Continue registering?");
+                builder.setCancelable(true);
+                builder.setPositiveButton("Yes", (DialogInterface.OnClickListener) (dialog, which) -> {
+                    registerButton.setBackgroundColor(0xFF808080);
+                    event.addEntrant(entrant);
+                    registerButton.setText("✔");
+                });
+                builder.setNegativeButton("No", (DialogInterface.OnClickListener) (dialog, which) -> {
+                    dialog.cancel();
+                });
+                AlertDialog geolocationWarning = builder.create();
+                geolocationWarning.show();
+
+            } else {
+                registerButton.setBackgroundColor(0xFF808080);
+                event.addEntrant(entrant);
+                registerButton.setText("✔");
             }
-        });
+        }
+    }
 
-        return view;
+    /**
+     * Toggles visibility of certain buttons depending on role of signedInAccount.
+     */
+    private void setPermissions() {
+        if (signedInAccount.getCurrentRole() == Account.AccountRole.admin) {
+            Log.d("EventFragment", "Admin user detected. Setting Admin permissions");
+            adminConstraint.setVisibility(View.VISIBLE);
+            registerButton.setVisibility(View.GONE);
+        } else {
+            Log.d("EventFragment", "Non-Admin user detected. Setting non-Admin permissions");
+            adminConstraint.setVisibility(View.GONE);
+            registerButton.setVisibility(View.VISIBLE);
+        }
```

Comment:
Wait so the event view from admin just uses the same as the user? That's pretty nice. Maybe just update in the header comment that Admin role uses this too just to make it clear